Documentation/linux-install-diagnosis.md

```diff
@@ -0,0 +1,144 @@
+# Linux package manager install failure diagnosis
+
+Before reporting a package manager install failure, check if your error message is on the list below. There are workarounds for some common failures, and for others, there are diagnosis steps you should run to reveal critical information.
+
+The Linux package manager install instructions are located at https://dotnet.microsoft.com/download.
+
+
+## `E: Unable to locate package dotnet-sdk-2.2`
+
+After running `apt-get install dotnet-sdk-2.2`, you may see:
+
+```
+$ apt-get install dotnet-sdk-2.2
+Reading package lists... Done
+Building dependency tree
+Reading state information... Done
+E: Unable to locate package dotnet-sdk-2.2
+E: Couldn't find any package by glob 'dotnet-sdk-2.2'
+E: Couldn't find any package by regex 'dotnet-sdk-2.2'
+```
+
+This is probably because the Microsoft package repository isn't set up correctly on your machine.
+
+The following commands give more info about the error and the current state of your machine. Please run them and include the output in your issue report:
+
+```
+# This shows the package repositories you have configured. (Source: https://askubuntu.com/a/741948)
+grep -r --include '*.list' '^deb ' /etc/apt/sources.list /etc/apt/sources.list.d/
+
+# These commands show processor and Linux distribution information.
+uname -a
+cat /etc/os-release
+
+# You can copy this entire block of text and paste it into your terminal for convenience.
+```
+
+Next are common issues and resolutions:
+
+### Attempting to use the package repository on a non-x64 device
+The Microsoft package repository only works on x64 (`x86_64`). For example, a Raspberry Pi uses an ARM architecture, so you can't install .NET Core on that device using the package repository.
+
+In this case, you can [install from binaries](#install-from-binaries).
+
+### Package feed not installed due to bug in `packages-microsoft-prod.deb`
+The output from the commands above should include a line like this:
+
+```
+/etc/apt/sources.list.d/microsoft-prod.list:deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.10/prod cosmic main
+```
+
+First, try reinstalling the package to work around a bug in an old version that improperly handles upgrades:
+
+```
+sudo dpkg --purge packages-microsoft-prod && sudo dpkg -i packages-microsoft-prod.deb
```

Comment:
The deb will probably need to be downloaded (or found) before attempting to install.